File: 简单工厂模式/模式实现/Api_SimFactory.java
interface Api {
	public void operation(String s);
}

class ImplA implements Api {
	public void operation(String s){
		System.out.println("ImlA s == " +s );
	}
}


class ImplB implements Api {
	public void operation(String s){
		System.out.println("ImplB s == " +s );
	}
}

class Factory {
	
	public static Api createApi(int condition){
		Api api = null;

		if(condition == 1){
			api = new ImplA();
		}else{
			api = new ImplB();
		}
		return api;
	}
}

public class Api_SimFactory {
	public static void main(String[] args){
		Api api = Factory.createApi(1);
		api.operation("simple factory !");
	}
}

File: 简单工厂模式/模式实现/Api_SimFactory.php
<?php 
interface Api {
	public function operation($s);
}

class ImplA implements Api {
	public function operation($s){
		echo "ImlA s == $s";
	}
}

class ImplB implements Api {
	public function operation($s){
		echo "ImplB s == $s";
	}
}

class Factory {

	public static function createApi($i=1){
		$a = null;
		if($i == 1){
			$a = new ImplA();
		}else{
			$a = new ImplB();
		}
		return $a;
	}
}

$api = Factory::createApi(1);
$api->operation("implA");